chat-agent/tests/test_system_prompt_security.py

```
@@ -0,0 +1,729 @@
+    @pytest.mark.asyncio
+    async def test_refuse_python_code_request(self, mock_env_vars, mocker):
+        """Test that Python code writing requests are refused."""
+        mock_llm = AsyncMock()
+        mock_llm.bind_tools = MagicMock(return_value=mock_llm)
+        mock_llm.ainvoke = AsyncMock(return_value=AIMessage(content="I'm sorry, I cannot help with writing code. I'm here to assist with Meals, Guest WiFi, and Leave management tasks only."))
+
+        with patch("agent.agent.ChatOpenAI", return_value=mock_llm):
+            with patch("agent.agent.get_employee_location", return_value=None):
+                response = await run_agent(
+                    user_message="Write a Python function to sort a list",
+                    access_token="test-token",
```

Ruff (0.15.10)

[error] Possible hardcoded password assigned to argument: "access_token"

(S106)